pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+                                added connection.
+    :type device_to_activate: str
+    """
+    con = client.add_connection_finish(result)
+    log.debug("connection %s added:\n%s", con.get_uuid(),
+              con.to_dbus(NM.ConnectionSerializationFlags.ALL))
+    if device_to_activate:
+        device = client.get_device_by_iface(device_to_activate)
+        if device:
+            log.debug("activating with device %s", device.get_iface())
+        else:
+            log.debug("activating without device specified (not found)")
+        client.activate_connection_async(con, device, None, None)
+
+
+def create_slave_connection(slave_type, slave_idx, slave, master, settings=None):
+    """Create a slave NM connection for virtual connection (bond, team, bridge).
+
+    :param slave_type: type of slave ("bond", "team", "bridge")
+    :type slave_type: str
+    :param slave_idx: index of the slave for naming
+    :type slave_idx: int
+    :param slave: slave's device name
+    :type slave: str
+    :param master: slave's master device name
+    :type master: str
+    :param settings: list of other settings to be added to the connection
+    :type settings: list(NM.Setting)
+
+    :return: created connection
+    :rtype: NM.SimpleConnection
+    """
+
+    settings = settings or []
+    slave_name = "%s slave %d" % (master, slave_idx)
+
+    con = NM.SimpleConnection.new()
+    s_con = NM.SettingConnection.new()
+    s_con.props.uuid = NM.utils_uuid_generate()
+    s_con.props.id = slave_name
+    s_con.props.slave_type = slave_type
+    s_con.props.master = master
+    s_con.props.type = '802-3-ethernet'
+    # HACK preventing NM to autoactivate the connection
+    # The real network --onboot value (ifcfg ONBOOT) will be set later by
+    # update_onboot
+    s_con.props.autoconnect = False
+    con.add_setting(s_con)
+
+    s_wired = NM.SettingWired.new()
+    con.add_setting(s_wired)
+
+    for setting in settings:
+        con.add_setting(setting)
+
+    return con
+
+
+def is_infiniband_device(nm_client, device_name):
+    """Is the type of the device infiniband?"""
+    device = nm_client.get_device_by_iface(device_name)
+    if device and device.get_device_type() == NM.DeviceType.INFINIBAND:
+        return True
+    return False
+
+
+def bound_hwaddr_of_device(nm_client, device_name, ifname_option_values):
+    """Check and return mac address of device bound by device renaming.
+
+    For example ifname=ens3:f4:ce:46:2c:44:7a should bind the device name ens3
+    to the MAC address (and rename the device in initramfs eventually).  If
+    hwaddress of the device devname is the same as the MAC address, its value
+    is returned.
+
+    :param devname: device name
+    :type devname: str
+    :param ifname_option_values: list of ifname boot option values
+    :type ifname_option_values: list(str)
+    :return: hwaddress of the device if bound, or None
+    :rtype: str or None
+
```

Comment:
Could you please remove this blank line.

Reply:
https://github.com/rhinstaller/anaconda/pull/1771/commits/ba752145484a43443c54981bbcfad4dc36696a16